cypress/integration/layout.cy.js

```diff
@@ -0,0 +1,103 @@
+//  TODO: Test drag-and-drop in the layout
+
+import {
+    DIMENSION_ID_DATA,
+    DIMENSION_ID_PERIOD,
+    DIMENSION_ID_ASSIGNED_CATEGORIES,
+    AXIS_ID_COLUMNS,
+    AXIS_ID_FILTERS,
+    getFixedDimensions,
+    getDynamicDimensions,
+} from '@dhis2/analytics'
+import {
+    expectDimensionModalToBeVisible,
+    clickDimensionModalHideButton,
+} from '../elements/DimensionModal'
+import {
+    openDimension,
+    openContextMenu,
+    clickContextMenuAdd,
+    clickContextMenuRemove,
+    clickContextMenuMove,
+    expectDimensionToHaveSelectedStyle,
+    clickContextMenuDimSubMenu,
+    expectAxisToHaveDimension,
+    expectAxisToNotHaveDimension,
+    expectLayoutToNotHaveDimension,
+} from '../elements/Layout'
+import { goToStartPage } from '../elements/StartScreen'
+
+const TEST_FIXED_DIMS = Object.values(getFixedDimensions())
+const TEST_DYNAMIC_DIMS = Object.values(getDynamicDimensions())
+
+describe('interacting with the dimensions panel', () => {
+    it('navigates to the start page', () => {
+        goToStartPage()
+    })
+    describe('clicking layout items', () => {
+        TEST_FIXED_DIMS.forEach(dim => {
+            describe(`${dim.name}`, () => {
+                it('clicks the dimension', () => {
+                    openDimension(dim.id)
+                })
+                it('shows the dimension modal', () => {
+                    expectDimensionModalToBeVisible(dim.id)
+                })
+                it('clicks the modal hide button', () => {
+                    clickDimensionModalHideButton()
+                })
```

Comment:
To me this is a good candidate for consolidating into a single test "Data dimension modal opens and closes".

Also, these tests have a duplicated assertion `expectDimensionModalToBeVisible` (check `openDimension` function)

Reply:
Yes I agree. Initially I thought it would be good if each action could self-evaluate (e.g. `openDimension` checking if dimension is open), but I realised it led to a lot of duplication. I'll go through the earlier tests and refactor this.

The line 39 `describe` acts as a wrapper to get a list of dimension names in the test tree structure, but I guess suffixing the description with the dimension name, like `clicking layout items - Data` achieves the same thing with less code.

Reply:
Fixed 👍